packages/mui-material/src/Radio/Radio.js

```diff
@@ -35,26 +35,32 @@ const RadioRoot = styled(SwitchBase, {
     return [styles.root, styles[`color${capitalize(ownerState.color)}`]];
   },
 })(({ theme, ownerState }) => ({
-  color: theme.palette.text.secondary,
+  color: (theme.vars || theme).palette.text.secondary,
   '&:hover': {
-    backgroundColor: alpha(
-      ownerState.color === 'default'
-        ? theme.palette.action.active
-        : theme.palette[ownerState.color].main,
-      theme.palette.action.hoverOpacity,
-    ),
+    backgroundColor: theme.vars
+      ? `rgba(${
+          ownerState.color === 'default'
+            ? theme.vars.palette.action.active
```

Comment:
```suggestion
            ? theme.vars.palette.action.activeChannel
```
We need here the `activeChannel`, similar to how we use the `mainChannel`. However, at this moment we are note generating it. I will push an update later to resolve this.

Reply:
@alisasanib if you want to try this yourself, take a look at how it is done in https://github.com/mui/material-ui/pull/32519 (the `dividerChannel` is added in that PR, but it should be very similar :))

Reply:
Oh thanks for the explanation! Yes I would like to work on it. Is it going to be sth like this:

```
export interface PaletteWithChannels {
  ...
  action: TypeAction & { disabledChannel: string; activeChannel: string };
  ...
}
```

Reply:
Yes and we need to also add the channel color